docs/instrumentation-libraries.md

```diff
@@ -0,0 +1,99 @@
+# Support for OpenTelemetry Instrumentation Libraries
+
+OpenTelemetry instrumentation are implemented using a variety of techniques.
+Sometimes the instrumentation is built directly into the library being
+instrumented, sometimes the instrumentation requires the addition of a separate
+instrumentation library, and other times it is a hybrid of those approaches. This
```

Comment:
Add links to the spec where these terms are defined. For example https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-library.